### DISCUSSION TITLE
Streamline Choice dropdown menu and Macro Management in settings

### DISCUSSION THREAD

**Discussion by sarahscheffler** [chhoumann/quickadd#716] [General]
I'm new to QuickAdd.  I love the macro features.  I also found them very difficult to use in part due to some baffling UI choices.  I'm going to walk through a surprisingly-difficult macro creation process I went through as a newbie, and then suggest changes.

To illustrate what I mean, here's what I just went through to make what I thought was a simple macro (open my Daily Note, make a newline, go to the newline, make a new Task).  To make a new macro from a fresh install, you are presented with the following in the QuickAdd settings:

![quickadd_fresh](https://github.com/user-attachments/assets/36881266-5526-4a6e-bc3f-98ac24d2758b)

The process of creating a new macro is surprisingly difficult for a new user.  (Yes, I read the Getting Started page and the MacroChoice page.  Neither was helpful, more on that below.)  The process is:
- Do not click "Manage Macros" as one might expect (you _can_ make macros from the Manage Macros menu, but you won't be able to add them to the command palette, this took me a bit to figure out)
- Instead you have to click "Template".  This looks like a button rather than a dropdown menu.  (This isn't my theme -- I'm using minimal and plenty of other dropdown menus have the little up-down arrows -- just this one doesn't.)  Especially because it looks like a button, you might not think to click this, since you don't want a template, so it will be very unclear how to actually make a macro that appears in the settings.
- Once you click Template, you can change it to Macro -- okay fine.
- Then you might click "Add Choice"
- (if you didn't read the docs to know that a macro is a kind of choice, this might confuse you, since in regular speech I wouldn't think of a macro as a choice)
- But anyway if you do click this, a hard-to-see error pops up in the top right saying "Choice Name Is Invalid".
- Eventually you realize you have to name your macro now, rather than during the pop-up.  So you name the macro.
- Then you click Add Choice and you can finally edit the macro by clicking on the little gear.
- You can also enable the macro in command palette by clicking the little lightning bolt, which wasn't obvious and I found out by googling.
- If at any point during this you check the [macros documentation](https://quickadd.obsidian.guide/docs/Choices/MacroChoice), none of these questions are answered, some of it is clearly out of date, and what is there at times is actively misleading compared to the real process.

Rather than this, I suggest the following UI changes:
- Either remove the "Manage Macros" button or make it able to create a full new macro that will appear in the choice list and be callable from command palette, rather than an unusable macro relegated to this weird half-accessible space.
- Change the "Template/Capture/Macro/Multi" thing to actually be a dropdown menu instead of whatever it is now
- If you click Add Choice before you name the choice, have a little pop-up appear to name the choice rather than just erroring.
- Make a toggle for the macro to appear in command palette in the macro-editing page itself, in case you don't realize offhand what the lightning bolt means.
- Fix the [MacroChoice](https://quickadd.obsidian.guide/docs/Choices/MacroChoice) documentation page to reflect the real process of creating a macro.

And unrelated to this, I also suggest referencing the [Code Editor Shortcuts](https://github.com/timhor/obsidian-editor-shortcuts) plugin somewhere in the documentation here as a suggested option, as it took me quite a long time to find a plugin that could move the cursor or add a newline from command palette.  (It seems captures do not play well with newlines, even if incorporating newlines from templates, which was very frustrating.)  The general code editor shortcuts seem like a good synergy with QuickAdd for improved automation.

In any case, thank you for listening, and thanks for the plugin, which I hope can become better with these small changes.  I realize I'm new at this, but I think this would really help new adoptees of the plugin.  Thanks!

**Comment by chhoumann** [comment]
Coming back to this one very late - sorry about that, and thank you for the unusually thorough writeup. Feedback at this level of detail is rare, and essentially every point has since been addressed in the settings rework:

- **"Manage Macros" half-space**: removed entirely. Macros now live directly on the Macro choice - you add a Macro choice and build its commands right there, so the orphaned, half-accessible macro space is gone.
- **The type selector that looked like a button**: reworked. "Add choice" now opens a menu listing each type with an icon, a label, and a one-line description.
- **"Choice Name Is Invalid"**: gone. New choices are created with a sensible default name and open for configuration right away, so there's no name-first error dead end.
- **The lightning bolt**: still on the choice row, but it now has a proper "Command palette" tooltip and an accessible pressed state, so you can tell what it does without googling.
- **MacroChoice docs**: rewritten from scratch, with the real step-by-step creation flow, screenshots, and a reference for every command type: https://quickadd.obsidian.guide/docs/Choices/MacroChoice
- **Cursor movement**: macros now ship built-in Editor commands, including "Move cursor to file start/end" and "Move cursor to line start/end", so a flow like your daily-note task macro no longer needs a separate plugin for cursor moves. Capture's handling of newlines has also had several rounds of fixes since 2024.

If anything still feels rough on current QuickAdd, please open an issue - I'd genuinely like to hear it. Closing as resolved.

Again, thank you for this feedback! I really appreciate it.
